stan/math/prim/prob/wiener4_lccdf.hpp

```diff
@@ -0,0 +1,454 @@
+  } else {
+    return ret_t(log1m_exp(-exponent) - log1m_exp(2 * v * a));
+  }
+}
+
+/**
+ * Calculate parts of the partial derivatives for wiener_prob_grad_a and
+ * wiener_prob_grad_v (on log-scale)
+ *
+ * @tparam T_a type of boundary
+ * @tparam T_w type of relative starting point
+ * @tparam T_v type of drift rate
+ *
+ * @param a The boundary separation
+ * @param w_value The relative starting point
+ * @param v_value The drift rate
+ * @return 'ans' term
+ */
+template <typename T_a, typename T_w, typename T_v>
+inline auto wiener_prob_derivative_term(const T_a& a, const T_v& v_value,
+                                        const T_w& w_value) noexcept {
+  using ret_t = return_type_t<T_a, T_w, T_v>;
+  const auto exponent_m1 = log1p(-1.1 * 1.0e-8);
+  ret_t ans;
+  const auto v = -v_value;
+  const auto w = 1 - w_value;
+
+  if (fabs(v) == 0.0) {
+    return ret_t(-w);
+  }
+  if (v < 0) {
+    const auto exponent_with_1mw = 2.0 * v * a * (1.0 - w);
+    const auto exponent_with_w = 2 * a * v * w;
+    const auto exponent = 2 * a * v;
+
+    if (((exponent_with_1mw >= exponent_m1) || (exponent_with_w >= exponent_m1))
+        || (exponent >= exponent_m1)) {
+      return ret_t(-w);
+    }
+    ans = LOG_TWO + exponent_with_1mw - log1m_exp(exponent_with_1mw);
+    const auto diff_term = log1m_exp(exponent_with_w) - log1m_exp(exponent);
+    const auto log_w = log(w);
+    if (log_w > diff_term) {
+      ans += log_diff_exp(log_w, diff_term);
+      ans = exp(ans);
+    } else {
+      ans += log_diff_exp(diff_term, log_w);
+      ans = -exp(ans);
+    }
+  } else {
+    const auto exponent_with_1mw = -2.0 * v * a * (1.0 - w);
+    const auto exponent = (-2 * a * v);
+    if ((exponent_with_1mw >= exponent_m1) || (exponent >= exponent_m1)) {
+      return ret_t(-w);
+    }
+    ans = LOG_TWO - log1m_exp(exponent_with_1mw);
+    const auto diff_term
+        = log_diff_exp(exponent_with_1mw, exponent) - log1m_exp(exponent);
+    const auto log_w = log(w);
+    if (log_w > diff_term) {
+      ans += log_diff_exp(log_w, diff_term);
+      ans = -exp(ans);
+    } else {
+      ans += log_diff_exp(diff_term, log_w);
+      ans = exp(ans);
+    }
+  }
+  if (fabs(ans) < INFTY) {
+    return ans;
+  } else {
+    return ret_t(NEGATIVE_INFTY);
+  }
+  return ans;
+}
+
+/**
+ * Calculate the derivative of the wiener probability w.r.t. 'a' (on log-scale)
+ *
+ * @tparam T_a type of boundary
+ * @tparam T_w type of relative starting point
+ * @tparam T_v type of drift rate
+ *
+ * @param a The boundary separation
+ * @param w The relative starting point
+ * @param v The drift rate
+ * @return Gradient w.r.t. a
+ */
+template <typename T_a, typename T_w, typename T_v>
+inline auto wiener_prob_grad_a(const T_a& a, const T_v& v,
+                               const T_w& w) noexcept {
+  using ret_t = return_type_t<T_a, T_w, T_v>;
+  if (fabs(v) == 0.0) {
+    return ret_t(0.0);
+  }
```

Comment:
What are the odds `v` will be exactly 0? I think this path is unlikely and would remove it.

In general, the odds a parameter takes on any given single value is very low so I would remove if statements like this. 99.9% of the time they won't be taken and forces the compiler to do a check and branch that is often unnecessary

Reply:
The user can call the model with a fixed value for v. In some cases the user could want to set `v=0` for his tests. Then, `v` will not be estimated and is constant at 0. For this case we need these if-statements.

Reply:
Would a value of  0 for v cause the rest of the code to error out? If not then remove these if statements

Reply:
I removed all spots where it is not necessary. One spot is still needed in `wiener_prob` to not cause an error.